framework/src/controller/controller.js

```diff
@@ -93,54 +93,65 @@ module.exports = class Controller {
 		);
 	}
 
-	async _loadModules() {
-		return Promise.each(Object.keys(this.modules), m =>
-			this._loadInMemoryModule(m)
+	async _loadModules(modules) {
+		return Promise.each(Object.keys(modules), alias =>
+			this._loadInMemoryModule(
+				alias,
+				modules[alias].klass,
+				modules[alias].options
+			)
 		);
 	}
 
-	async _loadInMemoryModule(alias) {
-		const module = this.modules[alias].spec;
-		const moduleConfig = this.modules[alias].config;
+	async _loadInMemoryModule(alias, Klass, options) {
+		const module = new Klass(options);
 		validateModuleSpec(module);
 
+		const moduleAlias = module.constructor.alias;
+		const moduleInfo = module.constructor.info;
```

Comment:
We could deconstruct the object here.

```suggestion
		const {name, version} = module.constructor.info;
```